firefox-ios/Client/Frontend/Extensions/DevicePickerViewController.swift

```diff
@@ -122,45 +119,42 @@ class DevicePickerViewController: UITableViewController {
+            return
+        }
 
-            self.devices = state.remoteDevices.compactMap { device in
-                guard device.capabilities.contains(.sendTab) else { return nil }
-
-                let typeString = "\(device.deviceType)"
-                let lastAccessTime = device.lastAccessTime == nil ? nil : UInt64(clamping: device.lastAccessTime!)
-                return RemoteDevice(id: device.id,
-                                    name: device.displayName,
-                                    type: typeString,
-                                    isCurrentDevice: device.isCurrentDevice,
-                                    lastAccessTime: lastAccessTime,
-                                    availableCommands: nil)
-            }
+        let currentIds = self.devices.map { $0.id ?? "" }.sorted()
+        let newIds = state.remoteDevices.map { $0.id }.sorted()
+        if !currentIds.isEmpty, currentIds == newIds {
+            return
+        }
 
-            if self.devices.isEmpty {
-                self.navigationItem.rightBarButtonItem = nil
-            } else {
-                self.navigationItem.rightBarButtonItem = UIBarButtonItem(title: .SendToSendButtonTitle,
-                                                                         style: .done,
-                                                                         target: self,
-                                                                         action: #selector(self.send))
-                self.navigationItem.rightBarButtonItem?.isEnabled = false
-            }
+        self.devices = state.remoteDevices.compactMap { device in
+            guard device.capabilities.contains(.sendTab) else { return nil }
+
+            let typeString = "\(device.deviceType)"
+            let lastAccessTime = device.lastAccessTime == nil ? nil : UInt64(clamping: device.lastAccessTime!)
+            return RemoteDevice(id: device.id,
+                                name: device.displayName,
+                                type: typeString,
+                                isCurrentDevice: device.isCurrentDevice,
+                                lastAccessTime: lastAccessTime,
+                                availableCommands: nil)
+        }
 
-            self.loadingState = .loaded
-            self.tableView.reloadData()
+        if self.devices.isEmpty {
+            self.navigationItem.rightBarButtonItem = nil
+        } else {
+            self.navigationItem.rightBarButtonItem = UIBarButtonItem(title: .SendToSendButtonTitle,
+                                                                     style: .done,
+                                                                     target: self,
+                                                                     action: #selector(self.send))
+            self.navigationItem.rightBarButtonItem?.isEnabled = false
         }
+
+        self.loadingState = .loaded
+        self.tableView.reloadData()
```

Comment:
Changes here are all around removing the `startup` call. Startup should happen only once in the process, but it was happening a lot because we already started up at the beginning of the ShareTo extension, then here we listen to an Notification and trigger `loadList` which creates many Firefox Account objects, when there should only be one.

The whole way extensions (both this one and NotificationService) access the accounts objects could use some love and refactoring